Fix placement of Insights advisor docs (#3910)
theforeman/foreman-documentation: guides/common/modules/con_overview-of-project-integration-with-insights.adoc

[id="overview-of-{project-context}-integration-with-insights"]
= Overview of {Project} integration with Insights

ifdef::installing-satellite-server-connected[]

> In a connected {Project} environment, you can register your {ProjectServer} with Insights or deploy the local {advisorengine}.

Reviewer: ```suggestion
In a connected {Project} environment, you can register your {ProjectServer} with Insights or deploy {advisorengine} locally.
```

endif::[]
ifdef::installing-satellite-server-disconnected[]

> In a disconnected {Project} environment, you can install the local {advisorengine}.

Reviewer: ```suggestion
In a disconnected {Project} environment, you can install {advisorengine} locally.
```

endif::[]

ifdef::installing-satellite-server-connected[]
{ProjectServer} as an Insights client::

> Connecting your {ProjectServer} to Insights as a client enables you to use the Insights dashboard to quickly identify key risks to stability, security, and performance.

Reviewer: update: Reading the preview, it now seems more like this refers to hosted Insights and not local.

Should we also call out here that using non-local (hosted) Insights gives you access to all Insights services, including Advisor, Vulnerability, Subscriptions, etc.? This would be a good contrast to local Insights where you only have Advisor.

+
Note that you do not require a Red{nbsp}Hat Insights entitlement in your subscription manifest.
endif::[]

The {advisorengine}::

> The {advisorengine} analyzes system health and configuration by applying predefined rules to a small set of local data, such as installed packages, running services, and configuration settings.

Reviewer: This renders as "The Insights advisor in Satellite." I don't think the "the" is necessary in most uses.

```suggestion
{advisorengine} analyzes system health and configuration by applying predefined rules to a small set of local data, such as installed packages, running services, and configuration settings.
```

Perhaps {advisorengine} could be renamed to {insights_advisor_in_satellite} to discourage unnecessary articles in the future.

When you deploy the Insights advisor engine locally on {Project}, you can generate Insights recommendations without sending system data to Red{nbsp}Hat services.
+

> Note that with the {advisorengine} enabled, you cannot use the hosted Subscriptions service.

Reviewer: ```suggestion
Note that with {advisorengine} enabled, you cannot use the hosted Subscriptions service.
```

> Enabling the {advisorengine} prevents you from using any Red{nbsp}Hat Hybrid Cloud Console services.

Reviewer: ```suggestion
Enabling {advisorengine} prevents you from using any Red{nbsp}Hat Hybrid Cloud Console services.
```

nice, thanks for including this :)

[id="overview-of-{project-context}-integration-with-insights"]
= Overview of {Project} integration with Insights

ifdef::installing-satellite-server-connected[]
In a connected {Project} environment, you can use the hosted Insights services or you can install {insights-advisor} locally.
endif::[]
ifdef::installing-satellite-server-disconnected[]
In a disconnected {Project} environment, you can install {insights-advisor} locally.
endif::[]

ifdef::installing-satellite-server-connected[]
Hosted Insights services::

> Registering your {ProjectServer} with Insights gives you access to all Insights services, including the Advisor, Vulnerability, and Subscriptions services.

Reviewer: I am still not sure this is technically accurate.

I think to use hosted Insights, you aren't required to connect Satellite at all, but you will not get any reporting about Satellite-registered hosts.

To use hosted Insights with Satellite-registered hosts, I think you could 
- allow inventory uploads (this allows hosts to be reported in the daily rh_cloud reports and be included in Subscriptions service reporting. It also allows the hosted Advisor and Vulnerability service to provide recommendations.)
- configure the cloud connector (this allows you to remediate recommendations using the hosted Advisor service to registered Satellite hosts)

Both of these are optional, and **neither** (to my knowledge) require "registering your {ProjectServer} with Insights" or installing insights-client on your Satellite base system. But @ehelms @parthaa @chris1984 please fill in any info I am misremembering here.

Author: For context: I was working with the existing wording we have here: https://docs.redhat.com/en/documentation/red_hat_satellite/6.17/html-single/installing_satellite_server_in_a_connected_network_environment/index#using-insights-with-satellite-server_satellite  This is our existing description of the non-local Insights functionality.

Reviewer 2: That particular section is about being able to manage the Satellite itself with Insights. And not about what capabilities you are enabling for clients connected to Satellite.

Author: Lines 12-15 are supposed to summarize the "non-local" Insights feature that has been available for a long time: `satellite-maintain packages install insights-client` + satellite-installer --register-with-insights` (currently documented in https://docs.theforeman.org/nightly/Installing_Server/index-satellite.html#using-insights-with-satellite-server_satellite -- on master). If this isn't "hosted Insights", then what do we want to call it?

[5.1.1. Overview of Satellite integration with Insights](https://theforeman-foreman-documentation-preview-pr-3910.surge.sh/nightly/Installing_Server/index-satellite.html#overview-of-satellite-integration-with-insights) (on this branch) shows the intention: To explain to users what options they have for integration with Insights. Does seeing "hosted Insights" in context like this help clarify what I'm looking for?

Reviewer: As it stands in the link you posted, we basically say

```
5.1.1. Overview of Satellite integration with Insights
[introductory blurb]

Hosted Insights services
[describe hosted Insights]

Insights advisor in Satellite
[describe Insights advisor in Satellite]

5.1.2. Enabling Insights on your Satellite Server
[describe installing and enabling insights-client on the Satellite base system, which has nothing to do with either of the above]
5.1.3. Disabling Red Hat Insights registration
[describe undoing the thing we did in 5.1.2]
```

I think 5.1.2 and 5.1.3 either need to go away completely, or be in a completely different place, so that it's not implied that they are related to 5.1.1.
